sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -221,3 +223,22 @@ object FileSourceStrategy extends Strategy with Logging {
     case _ => Nil
   }
 }
+
+/**
+ * Extractor that handles `CoalesceBuckets` in the child plan extracted from `ScanOperation`.
+ */
+object ScanOperationWithCoalescedBuckets {
```

Comment:
plz add tests for this new extractor, too?

Reply:
I added it in CoalesceBucketsInEquiJoinSuite.scala. (Please let me know if it makes more sense to have it in FileSourceStrategySuite.scala)